Button/Button.module.less

```diff
@@ -305,4 +306,26 @@
 			}
 		});
 	}
+
+	&:global(.spottable):active,
+	:global(.spottable):active & {
```

Comment:
We'll likely want to make a mixin that does this, so the component LESS files can stay more focused on styling the visual states. We have a `.focus()` mixin that will be extremely similar to the implementation of this.

Also, typically we'll put state rules inside the `.applySkins()` mixin for predictability and correctness of the output selectors in complex scenarios.

Reply:
Here I need some guidance. the focus() mixin comes from enact/spotlight. Should there be also an .active() mixin? What about the similar implementation, but  for focus state from this button.module.less. Should it be moved also?
I will add an agate-button-active mixin in agate mixins.less for the moment. Looking forwatd for your feedback

Reply:
Yes, `enact/spotlight` probably should have an `active` mixin added (at some point). Possibly even a `.visited` but the value of that may be pretty low.

Regarding the focus mixin, in this case, I don't think we actually need the elaborateness of the selector-set. One of the lines refers to any parent that has focus, that also contain a button, the other line refers to if this button is focused. We don't typically put focusable things inside focusable things, so this is probably overkill. A simpler `.focus({ rules:here; });` usage would probably be satisfactory. Also, it should live inside the `applySkins` block after the other skin rules so it can take precedence.

The new "active" mixin is pretty specific to button. We could rewrite it to be more like spotlight's existing mixins and have it accept a ruleset. Then the component (any component) can add its own rules. Here's an example (altered focus, taken from spotlight):

```css
// Focused Spottable elements
.active(@rules; @target) when (isruleset(@rules)) and (@target = parent) {
	.spottable({
		&:active {
			@rules();
		}
	}, @target);
}
.active(@rules) when (isruleset(@rules)) {
	.spottable({
		&:active {
			@rules();
		}
	});
}
```

And here's how it would be used:
```css
.button {
	...
	.applySkins({
		...

		.active({
			.bg {
				background-color: @agate-button-active-bg-color;
				background-image: @agate-button-active-bg-image;
				border: @agate-button-border-width solid @agate-button-border-color;;
			}

			&,
			.client {
				color: @agate-button-active-color;
			}

			&.highlighted {
				.bg {
					opacity: 1;
				}
			}
		});
	});
}
```